### ISSUE TITLE
documentation for relion convention

### ISSUE THREAD

**Issue by geoffwoollard** [mjo22/cryojax#116]
@mjo22 I think I figured out the correspondence of cryojax to Relion:

Note: 
This matched with relion star file conventions, where data has a star file loaded in:
shot_info_view_phi -> _rlnAngleRot
shot_info_view_theta -> _rlnAngleTilt
shot_info_view_psi -> _rlnAnglePsi
shot_info_offset_x -> rlnOriginXAngst  (or info['rlnOriginX'] * pixel_size_angstrom)

```
volume_t = jnp.transpose(volume, axes=[2, 1, 0]) # NOTE TRANSPOSE
density = cs.VoxelGrid.from_density_grid(volume_t)
shape=density.weights.shape[:2]
pad_scale = 1

pixel_size = data['pixel_size']
manager = cs.ImageManager(shape=shape, pad_scale=pad_scale)
scattering = cs.FourierSliceExtract(manager, pixel_size=pixel_size)

pose = cs.EulerPose(view_phi=data['shot_info_view_phi'][idx],
                    view_theta=data['shot_info_view_theta'][idx],
                    view_psi=data['shot_info_view_psi'][idx],
                    offset_x=data['shot_info_offset_x'][idx],
                    offset_y=data['shot_info_offset_y'][idx],
                    convention='zxz', # NOTE CONVENTION IS NOT DEFAULT OF CRYOJAX
                    )
ensemble = cs.Ensemble(density=density, pose=pose)

optics = cs.CTFOptics(defocus_u=data['shot_info_defocus_u'][idx],
                    defocus_v=data['shot_info_defocus_v'][idx],
                    defocus_angle=data['shot_info_defocus_angle'][idx],
                    voltage=data['shot_info_voltage'][idx],
                    spherical_aberration=data['shot_info_spherical_aberration'][idx],
                    amplitude_contrast=data['shot_info_amplitude_contrast_ratio'][idx],
                    phase_shift=data['shot_info_phase_shift'][idx],
)
instrument_od = cs.Instrument(optics=optics)
pipeline = cs.ImagePipeline(scattering=scattering, instrument=instrument_od, ensemble=ensemble)
image = pipeline.render()

image = image[::-1] # NOTE MIRRORING
```

The image associated with the `idx` in the relion star file will match image

**Comment by mjo22** [comment]
Aha! This is awesome. Thanks for this. I had gotten things to match with cisTEM a while ago, so I assumed that they would be the same. My understanding was that cisTEM and relion use the same conventions, so there may be something fishy going on.

Note that I make a transpose here, line 264: https://github.com/mjo22/cryojax/blob/main/src/cryojax/simulator/density/_voxel_density.py. This transpose is

```python
density_grid = jnp.transpose(density_grid, axes=[2, 1, 0])
```

When this is undone, this will match the transpose you have here. I did this to match cisTEM 2D template matching. I know that both relion and cisTEM say that they use 'zyz' rotations, so we should figure out how to change things so we have ``convention = 'zyz'``. As for the mirroring... not sure how to fix this. I have x and y translations corresponding to a cartesian convention. Do you know if, when relion does translations, it really does so in logical i and j? Or does it do it in x and y?

Something else to try is to look into the code where I create coordinate systems: https://github.com/mjo22/cryojax/blob/main/src/cryojax/image/coordinates.py. See the function _make_coordinates_or_frequencies. What happens when you change the meshgrid call from `jnp.meshgrid(...)` to `jnp.meshgrid(..., indexing="ij")`?

We should make a branch to try and figure out how to get everything consistent.

**Comment by mjo22** [comment]
See the new unit test file called `test_agree_with_cistem.py`. I’ve gotten projection images to match cisTEM convention—I had done this a while ago but I’ve found some bugs recently that may have made things diverge. I think agreement with cisTEM should be the standard, but I’m curious what will happen if this relion test is redone. Maybe things will match now… if not, the solution will be to document the difference, as you have said here.

**Comment by geoffwoollard** [comment]
relion docs should be helpful
https://relion.readthedocs.io/en/release-3.1/Reference/Conventions.html

In compliance with the [Heymann, Chagoyen and Belnap (2005) standard](https://pubmed.ncbi.nlm.nih.gov/16043364/) relion uses a right-handed coordinate system with orthogonal axes X, Y and Z, where right-handed rotations are called positive, and Euler angles are defined as:

The first rotation is called rlnAngleRot and is around the Z-axis.

The second rotation is called rlnAngleTilt and is around the new Y-axis.

The third rotation is called rlnAnglePsi and is around the new Z axis

As such, relion uses the same Euler angles as XMIPP, SPIDER and FREALIGN.

The center of rotation of a 2D image of dimensions xdim x ydim is defined by ((int)xdim/2, (int)(ydim/2)) (with the first pixel in the upper left being (0,0). Note that for both xdim=ydim=65 and for xdim=ydim=64, the center will be at (32,32). This is the same convention as used in SPIDER and XMIPP. Origin offsets reported for individual images translate the image to its center and are applied BEFORE rotations.

Particle translations used to be in pixels (rlnOriginX and rlnOriginY) but this changed to Angstroms (rlnOriginXAngstrom and rlnOriginYAngstrom) in relion 3.1.

The unit of particle coordinates in a micrograph (rlnCoordinateX and rlnCoordinateY) is pixel in the aligned and summed micrograph (possibly binned from super-resolution movies). The origin is the first element in the 2D array of an MRC file. The origin is displayed at the upper-left corner in relion (other programs might display in other ways).

**Comment by mjo22** [comment]
Closing with #183. Conventions now match
